cunumeric/array.py

```diff
@@ -2994,6 +2994,31 @@ def reshape(self, shape, order="C") -> ndarray:
         --------
         Multiple GPUs, Multiple CPUs
         """
+
+        computed_shape = (shape,) if isinstance(shape, int) else shape
+
+        if sum(extent == -1 for extent in computed_shape) > 1:
+            raise ValueError("can only specify one unknown dimension")
+
+        unknown_extent = self.size
+        for extent in computed_shape:
+            if extent == -1 or extent == 0:
+                continue
+            if unknown_extent % extent != 0:
+                raise ValueError(
+                    f"cannot reshape array of size {self.size} "
+                    f"into shape ({extent})"
```

Comment:
I would give the full shape in this error message:
```
f"into shape {computed_shape}"
```

Reply:
this is to match NumPy's error message. I don't think the full shape is any more useful to identify the bug.